### PR TITLE
feat(lti): add the deep linking content picker

### PR DESCRIPTION
**Pull request by magic-peach**
Fixes part of #7408

 

### Screenshots of the UI changes (If any) -
Screenshot 2026-08-11 at 3 31 47 PM

This  demonstrates :
- The lti flag is on otherwise the page would display 404
- Both content types are offered including Circuits and Assignments as separate sections
- The listings are created from the signed in user's own data
---

## Code Understanding and AI Usage

**Did you use AI generated code (ChatGPT, Claude, Copilot, etc.) in any part of this PR? **
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

- What problem does your code solve?
Deep linking is the process where an instructor embeds a specific CircuitVerse content into a course like. a circuit or assignment rather than dropping in a link to the tool. When a platform sends a deep linking request, the tool has to show the instructor something to choose from. This is what this pr does

- What alternative approaches did you consider?
Build it as a ViewComponent: this is used in other places in Circuitverse but this is a single-purpose page with nothing to reuse yet so a component would add a class, template, and preview for no current benefit
Offer all public circuits with search: An instructor might want to embed a circuit from the library rather than their own work. I did not go with that because majorly the instructor would want their own circuits or assignments

- Why did you choose this specific implementation?
It displays exactly all the circuits and assignments created by the signed in user as a single option radio button to choose from all managed as long as the lti flag is kept on and verified with specs

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

- **New Features**
  - Added LTI deep-linking support for selecting and embedding circuits, assignments, and featured templates.
  - Added a picker displaying recently updated circuits and mentor-associated assignments.
  - Added localized instructions, selection controls, empty-state messages, and preserved configuration settings.
  - Added authentication and feature-access controls for the picker.

- **Bug Fixes**
  - Restricted displayed resources to authorized content and prevented duplicate selections.
  - Added ordering by recent updates for relevant content.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -146,6 +146,7 @@
     match 'launch', to: 'lti#launch', via: [:get, :post]
     get 'jwks', to: 'lti#jwks'
     get 'tool_config', to: 'lti#tool_config'
+    get 'deep_link', to: 'lti#deep_link'
```
Comment (anxkhn) [inline config/routes.rb:149]: form posts to `deep_link_path` but only get is routed, so embed 404s.
add post (and the action) here, or drop/disable the submit until that pr lands.

**Review by anxkhn — COMMENTED** [review]
@magic-peach thanks for the picker work. left one note on the embed path. happy to send a patch or hop on a call if that helps.
